tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -1317,3 +1317,150 @@ def test_spilu_0(self, dtype, xp, sp):
+        b = self._make_normalized_vector(dtype, xp)
+        if self.b_ndim == 2:
+            b = b.reshape(self.n, 1)
+        x0 = None
+        if self.x0 == 'ones':
+            x0 = xp.ones((self.n,), dtype=dtype)
+        atol = None
+        if self.atol == 'select-by-dtype':
+            atol = self._atol[dtype.char.lower()]
+        if atol is None and xp == numpy:
+            # Note: If atol is None or not specified, Scipy (at least 1.5.3)
+            # raises DeprecationWarning
+            with pytest.deprecated_call():
+                return sp.linalg.cgs(a, b, x0=x0, M=M, atol=atol)
+        else:
+            return sp.linalg.cgs(a, b, x0=x0, M=M, atol=atol)
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    def test_dense(self, dtype, xp, sp):
+        a, M = self._make_matrix(dtype, xp)
+        if self.use_linear_operator:
+            a = sp.linalg.aslinearoperator(a)
+            if M is not None:
+                M = sp.linalg.aslinearoperator(M)
+        return self._test_cgs(dtype, xp, sp, a, M)
+
+    @pytest.mark.parametrize('format', ['csr', 'csc', 'coo'])
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    def test_sparse(self, format, dtype, xp, sp):
+        if runtime.is_hip and format == 'csc':
+            pytest.xfail('may be buggy')  # trans=True
+
+        a, M = self._make_matrix(dtype, xp)
+        a = sp.coo_matrix(a).asformat(format)
+        if self.use_linear_operator:
+            a = sp.linalg.aslinearoperator(a)
+        if M is not None:
+            M = sp.coo_matrix(M).asformat(format)
+            if self.use_linear_operator:
+                M = sp.linalg.aslinearoperator(M)
+        return self._test_cgs(dtype, xp, sp, a, M)
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    def test_empty(self, dtype, xp, sp):
+        if not (self.x0 is None and self.M is None and self.atol is None and
+                self.use_linear_operator is False):
+            raise unittest.SkipTest
+        a = xp.empty((0, 0), dtype=dtype)
+        b = xp.empty((0,), dtype=dtype)
+        if self.atol is None and xp == numpy:
+            # Note: If atol is None or not specified, Scipy (at least 1.5.3)
+            # raises DeprecationWarning
+            with pytest.deprecated_call():
+                return sp.linalg.cgs(a, b)
+        else:
+            return sp.linalg.cgs(a, b)
+
+    @testing.for_dtypes('fdFD')
+    def test_callback(self, dtype):
+        if not (self.x0 is None and self.M is None and self.atol is None and
+                self.use_linear_operator is False):
+            raise unittest.SkipTest
+        xp, sp = cupy, sparse
+        a, M = self._make_matrix(dtype, xp)
+        b = self._make_normalized_vector(dtype, xp)
+        is_called = False
+
+        def callback(x):
+            print(xp.linalg.norm(b - a @ x))
```

Comment:
Would you remove this print?

Reply:
I added this print so that the test will have consistency with other tests.